src/wp-includes/class-wp-locale-switcher.php

```diff
@@ -103,14 +119,15 @@ public function switch_to_locale( $locale ) {
 	 * @return string|false Locale on success, false on failure.
 	 */
 	public function restore_previous_locale() {
-		$previous_locale = array_pop( $this->locales );
+		$previous_locale = array_pop( $this->stack );
 
 		if ( null === $previous_locale ) {
 			// The stack is empty, bail.
 			return false;
 		}
 
-		$locale = end( $this->locales );
+		$entry = end( $this->stack );
+		$locale = is_array( $entry ) ? $entry[0] : false;
```

Comment:
```suggestion
		$entry  = end( $this->stack );
		$locale = is_array( $entry ) ? $entry[0] : false;
```
Not sure how PHPCS didn't fail on this in CI 🤔. It produces a warning locally at least.